### PR TITLE
Import CUDAnative in Fields submodule

### LINKED ISSUE #634: Error when setting GPU field using CuArray
@arnscheidt reported this error when running this script

https://github.com/ali-ramadhan/antarctic-ice-shelf-meltwater-outflow/blob/eae17968349d72613ccbd0d5b856cc76b30b7850/ice_shelf_meltwater_outflow_2d.jl

```
InvalidIRError: compiling _set_gpu!(Cassette.Context{nametype(Ctx),Nothing,Nothing,GPUifyLoops.var"##PassType#422",Nothing,Cassette.DisableHooks}, typeof(Oceananigans.Fields._set_gpu!), OffsetArrays.OffsetArray{Float64,3,CUDAnative.CuDeviceArray{Float64,3,CUDAnative.AS.Global}}, CUDAnative.CuDeviceArray{Float64,3,CUDAnative.AS.Global}, RegularCartesianGrid{Float64,Oceananigans.Grids.Periodic,Oceananigans.Grids.Periodic,Bounded,StepRangeLen{Float64,Base.TwicePrecision{Float64},Base.TwicePrecision{Float64}}}) resulted in invalid LLVM IR
Reason: unsupported call to the Julia runtime (call to jl_f_apply_type)
```

Looking at the full stacktrace it looks like `threadIdx` is undefined so maybe we just forgot to import CUDAnative.jl in the Fields submodule following the big reorganization in #591. There should have been a test for this.

### PR DESCRIPTION
**Pull request by ali-ramadhan**
Fixes #634 

cc @arnscheidt

### CODE REVIEW

**Comment by ali-ramadhan** [comment]
Caught a file using CRLF line endings so I converted it to LF.

@glwagner Don't worry about the `test_fields.jl` diff, I only added this test:
https://github.com/climate-machine/Oceananigans.jl/pull/635/files#diff-8f5b49779286e5727c493613d03d2d1bR61-R67

File: src/Fields/set!.jl [hunk 1]
```diff
@@ -1,3 +1,4 @@
+using CUDAnative
```
Comment (glwagner) [inline src/Fields/set!.jl]: do we need `@hascuda` in front of this?

**Comment by ali-ramadhan** [comment]
> To be clear, the bug was a lack of import for `CUDAnative` (not in the function `set!` for `CuArray`)?

Yes. I've changed the PR title accordingly but left the issue title as is.

File: src/Fields/set!.jl (hunk 1 above)
Comment (ali-ramadhan) [reply src/Fields/set!.jl]: Hmmm, I think in the past we did but some versions ago all the CUDA packages became safe to import without a GPU. I guess we can clean up some uses of `@hascuda`.

Either way, tests are chugging along fine.

Comment (vchuravy) [reply src/Fields/set!.jl]: Not quite:

```
julia> using CUDAnative
[ Info: CUDAdrv.jl failed to initialize, GPU functionality unavailable (set JULIA_CUDA_SILENT or JULIA_CUDA_VERBOSE to silence or expand this message)
```

Comment (ali-ramadhan) [reply src/Fields/set!.jl]: Ah good point, I forgot about that warning (or info message). Hmmm, the fact that I've seen it probably means that there's some CUDA package import that needs a `@hascuda`.

Well, one approach is to `@hascuda` everything that might need it. Another is to show that warning to everyone.

Comment (glwagner) [reply src/Fields/set!.jl]: I vote `@hascuda`. We don't *require* CUDA, we simply support it. (It'd still be nice to emit an interpretable error if somebody tries to use `architecture=GPU()` without CUDA though.)

Comment (ali-ramadhan) [reply src/Fields/set!.jl]: :+1: We already do:
https://github.com/climate-machine/Oceananigans.jl/blob/4ef4d11eb95c2356847aeb4386e7e15e8d19fc9e/src/Models/incompressible_model.jl#L86-L88

**Pull request merged**